satpy/readers/geocat.py

```diff
@@ -56,7 +56,30 @@
 
 
 class GEOCATFileHandler(NetCDF4FileHandler):
-    """GEOCAT netCDF4 file handler."""
+    """GEOCAT netCDF4 file handler.
+
+    **Loading data with decode_times=True**
+
+    By default, this reader will use xarray_kwargs={"engine": "netcdf4", "decode_times": False}.
+    to match behavior of xarray when the geocat reader was first written.  To use different options
+    use reader_kwargs when loading the Scene:
+
+        scene = satpy.Scene(filenames,
+                            reader='geocat',
+                            reader_kwargs={'decode_times': True})
+    """
+
+    def __init__(self, filename, filename_info, filetype_info,
+                 **kwargs):
+        """Open and perform initial investigation of NetCDF file."""
+        kwargs.setdefault('xarray_kwargs', {}).setdefault(
+            'decode_times', False)
+        kwargs.setdefault('xarray_kwargs', {}).setdefault(
+            'engine', "netcdf4")
```

Comment:
Thinking about it, these can just be hardcoded can't they? So `kwargs.setdefault("xarray_kwargs", {"decode_times": False, "engine": "netcdf4"})`?

Reply:
At this point, I can't think of a use case where they could be anything else.

Reply:
If the geocat code changes, and the unit metadata becomes compliant, I think it would be nice to be able to interpret the scan line times when reading them.  I might be mistaken, but 75-78 as written would allow the user to set the decode_times back to True if it is possible to decode them?

Reply:
Yeah, it looks like it would work. My suggestion for doing the entire `xarray_kwargs` dictionary would also work, but would require users to make sure they include engine and decode_times if they were specifying xarray_kwargs at all. This is such a small thing and not sure it matters too much.

Reply:
Circling back to this again, sorry.... if the [kwargs are set as in a previous commit](https://github.com/joleenf/satpy/blob/b8a42dee445be65bc0db9e8a99feaf9cdd8e61c6/satpy/readers/geocat.py#L74-L78)

Doesn't that leave the engine as "netcdf4" if the user only enters a value for decode_times?  If no xarray_kwargs are provided, the values would be ```{"engine": "netcdf4", "decode_times": False}```?   

I think I am not understanding your point fully on this requested change to 
```kwargs.setdefault("xarray_kwargs", {"decode_times": False, "engine": "netcdf4"})```.

Reply:
You are correct. Yours is more flexible. That's why I said it doesn't matter much. My main reason for my suggestion was that it treated xarray_kwargs as an all-or-nothing keyword argument rather than individual things. Sorry for all the confusion for something so dumb (on my part). Let me know if you want to put it back the way you had it or leave it as is. Either way I'm fine with it and we can merge regardless of the pre-commit issue...which I think is a hiccup on pre-commit's side of things.

Reply:
It is fine.  I think either way works, so to avoid errors that can be introduced with too much more fiddling, I think I will leave it as is.